test/composertest.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+import sys
+import unittest
+
+# import Cockpit's machinery for test VMs and its browser test API
+sys.path.append(os.path.join(os.path.dirname(__file__), "../bots/machine"))
+import testvm # pylint: disable=import-error
+
+
+class ComposerTestCase(unittest.TestCase):
+    image = testvm.DEFAULT_IMAGE
+
+    def setUp(self):
```

Comment:
Some of the current tests need some sort of specific setup (changing the existing files used by composer) - I think this could be the right place to do this setup? So far I can think of two cases - changing sshd configuration to accept login with empty passwords and modifying a blueprint to contain custom kernel commandline parameters - please see https://github.com/weldr/lorax/blob/master/tests/test_cli.sh#L7.

Reply:
Yes and no :)

This `setUp` is called for all tests. Specific setup that should only happen for some of the tests should go into subclasses of this class (see test/check-cli for example, you could add a `setUp` there).

For now, the tests simply call out the `tests/cli/*.sh` scripts. We don't need to move anything over, unless it's easier to do in python.

Reply:
FTR @jikortus means we do modify some of the blueprints/kickstart files to make it easier for testing. For example live-iso by default forbids SSH connections because root account doesn't have a password there. 

Since this PR has already been merged we can work on fixing the remaining issues one by one. However be mindful that QE uses the existing bash scripts as they are to validate RPM builds so they must continue to work in stand-alone mode. Which likely means duplicating setup in some places (yikes).